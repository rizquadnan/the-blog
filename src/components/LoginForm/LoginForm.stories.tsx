import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { LoginForm } from './LoginForm'

type TLoginForm = React.ComponentProps<typeof LoginForm>
export default {
  title: 'UserInputs/LoginForm',
  component: LoginForm,
} as Meta<TLoginForm>

const Template: Story<TLoginForm> = (args) => <LoginForm {...args} />

export const Default = Template.bind({})
