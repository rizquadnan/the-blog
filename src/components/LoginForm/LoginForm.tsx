import { Input } from '@chakra-ui/input'
import { Button } from '@chakra-ui/button'
import {
  FormControl,
  FormErrorMessage,
  FormLabel,
} from '@chakra-ui/form-control'
import React from 'react'
import { useForm } from 'react-hook-form'
import { VStack } from '@/components/uiKit/VStack'

type TFormValues = {
  email: string
  password: string
}
type TLoginForm = {
  onSubmit: (props: TFormValues) => void
}

export function LoginForm(props: TLoginForm) {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<TFormValues>()

  return (
    <VStack as="form" minW="280px">
      <FormControl isInvalid={Boolean(errors.email)}>
        <FormLabel>Email</FormLabel>
        <Input
          {...register('email', { required: 'Email address is required' })}
          placeholder="Email"
          type="email"
        />
        <FormErrorMessage>{errors.email?.message}</FormErrorMessage>
      </FormControl>

      <FormControl isInvalid={Boolean(errors.password)}>
        <FormLabel>Password</FormLabel>
        <Input
          {...register('password', {
            required: 'Password is required',
            minLength: {
              value: 8,
              message: 'Password must be at least 8 characters',
            },
          })}
          placeholder="Password"
        />
        <FormErrorMessage>{errors.password?.message}</FormErrorMessage>
      </FormControl>
      <Button
        w="100%"
        mt="48px !important"
        onClick={handleSubmit(props.onSubmit)}
      >
        Submit
      </Button>
    </VStack>
  )
}

export default LoginForm
