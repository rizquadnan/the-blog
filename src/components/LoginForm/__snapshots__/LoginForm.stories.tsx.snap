// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Inputs/LoginForm Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Email",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Email",
      "pressed": undefined,
      "role": "textbox",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Password",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Password",
      "pressed": undefined,
      "role": "textbox",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Submit",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "user-inputs-loginform--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
